lib/Ast.ml

```diff
@@ -409,30 +423,40 @@ module Signature_item = struct
   let has_doc itm =
     match itm.psig_desc with
     | Psig_attribute atr -> Attr.is_doc atr
+    (* one attribute list *)
     | Psig_value {pval_attributes= atrs; _}
      |Psig_type (_, {ptype_attributes= atrs; _} :: _)
      |Psig_typesubst ({ptype_attributes= atrs; _} :: _)
      |Psig_typext {ptyext_attributes= atrs; _}
-     |Psig_modtype {pmtd_attributes= atrs; _}
-     |Psig_modtypesubst {pmtd_attributes= atrs; _}
-     |Psig_modsubst {pms_attributes= atrs; _}
      |Psig_open {popen_attributes= atrs; _}
      |Psig_extension (_, atrs)
      |Psig_class_type ({pci_attributes= atrs; _} :: _)
      |Psig_class ({pci_attributes= atrs; _} :: _) ->
         List.exists ~f:Attr.is_doc atrs
-    | Psig_recmodule
-        ( {pmd_type= {pmty_attributes= atrs1; _}; pmd_attributes= atrs2; _}
-        :: _ )
+    (* two attribute list *)
+    | Psig_modtype
+        {pmtd_ext_attrs= {attrs_before= atrs1; attrs_after= atrs2; _}; _}
+     |Psig_modtypesubst
+        {pmtd_ext_attrs= {attrs_before= atrs1; attrs_after= atrs2; _}; _}
+     |Psig_modsubst
+        {pms_ext_attrs= {attrs_before= atrs1; attrs_after= atrs2; _}; _}
```

Comment:
Same as in the other module, I'd prefer not to unpack the `ext_attrs` here. This is going to grow and the existing rules need to stay readable.